six/modules/c++/six/include/six/ReadControl.h

```diff
@@ -108,12 +109,43 @@ class ReadControl
      *  This function reads in the image area specified by the region.
      *  If you want us to use a work buffer, you can set it through region.
      *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter
+     *
      *  Once read, the image buffer is set in both the region pointer,
      *  and in the return value, for convenience
+     *
+     *  For safety, prefer the overload below.
      */
     virtual UByte* interleaved(Region& region, size_t imageNumber) = 0;
 
     /*!
+     *  This function reads in the image area specified by the region.
+     *  If you want us to use a work buffer, you can set it through region.
+     *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter.
+     *
+     *  Once read, the image buffer is set in the passed buffer, and
+     *  also the region pointer and the return value for convenience.
+     *
+     * \param region Region stores row and cols to be read
+     * \param imageNumber Index of the image to read
+     * \param buffer Scoped array that holds the memory for the read-in image.
+     * This will be allocated by this function.
+     *
+     * \return Buffer of image data.  This is simply equal to buffer.get() and
+     * is provided as a convenience.
+     */
+    template<typename T>
+    UByte* interleaved(Region& region, size_t imageNumber,
+            mem::ScopedArray<T>& buffer)
+    {
+        buffer.reset(reinterpret_cast<T*>(interleaved(region, imageNumber)));
```

Comment:
I am not overfond of the `Region` holding any sort of pointer at all, as it makes it tricky to use the same `Region` for multiple reads.  If we didn't have to worry about breaking anything, I'd suggest removing it altogether. 

I'm trying to think through passing a Region that already has a buffer. I would hope that  the smart pointer parameter and the documentation would make it clear that that's what owns the memory at the end. Presumably the caller would already have it stored as a ScopedArray and not just a smart pointer, so that's what they'd end up passing as the third argument, and nothing would change.

Or if we change Region to hold a smart pointer, we don't even need the new overload, and there's one less pointer to the same data getting passed around.